app/screens/edit_post/edit_post.tsx

```diff
@@ -192,15 +238,21 @@ const EditPost = ({componentId, maxPostSize, post, closeButtonId, hasFilesAttach
             return;
         }
 
-        const res = await editPost(serverUrl, post.id, postMessage);
+        const currentFileIds = postFiles.map((file) => file.id).filter((id): id is string => Boolean(id));
```

Comment:
Why the guard (`id is string`) in the filter? Doesn't the type get properly set without it?

Reply:
`id` property in `FileInfo` is optional, which means its type is `string | undefined`. When we do 
```javascript
postFiles.map(file => file.id)
```
The return is an array of type `(string | undefined)[]`.
Without the type grad: `Boolean(id)` would filter out falsy values like `undefined`, but TypeScript would still think the result array is `(string | undefined)[]`. Even though the runtime filtering removes them.